docs/SOURCING-COMMITMENTS.md

```diff
@@ -0,0 +1,91 @@
+
+However, it's possible to be more impatient, although there are caveats - see the next section.
+
+### Sourcing external commitments
+
+In cases where you don't have enough utxos with valid commitments left, or you're not prepared to wait 5 blocks, there are alternatives made available. If you go into the `cmttools` directory you'll find a tool `add-utxo.py`. Run `python add-utxo.py --help` to see an explanation. In short, you can:
+
+* Add external utxos from a non-JM wallet, like Electrum or Core, one at a time, or from a prepared file (-r).
+* Add external utxos from another Joinmarket wallet (-w).
+* Delete existing external utxos (-d)
+* Validate utxos (-v, -o)
+
+Note that whatever utxo you add has to have a scriptPubKey of the **same** type as those of your current wallet (currently either nested or native segwit). This also corresponds to the type of orderbook you interact with.
+
+Also note that "adding a utxo" does **not** mean spending it! It only means adding a hash-value commitment and the pubkey, basically. Any Bitcoin utxo can be used as an effective 3-time usage token, without spending it.
+
+Be aware that Makers will see this utxo *only if* your usage is successful (they accept it and return their utxos in response), which is a reason not to use this "sourcing external" approach if you have any concerns about them getting that knowledge. Also, if you source a utxo from another non-JM wallet you will have to pass its private key in WIF compressed format to the script (it cannot be signed inside the wallet as this is not a standard Bitcoin ECDSA signature), in order to create the commitment - the usual strong warnings about handling private keys apply here, so use this approach only *very* carefully.
+
+### Commitment storage, and what if things go wrong.
+
+The commitments you've already used (just hash values) are stored in the `used` section of the file `cmttools/commitments.json` - **don't delete this file**. Although you won't need to read this file, it represents your memory of which commitments you've already used, so if you lose that record, your Taker scripts will find themselves spending a lot of time retrying commitments that the rest of the Joinmarket "network" knows are already used, and so will be rejected. This is only an inconvenience, but it could be pretty annoying. Any external commitments you sourced according to the previous section are also added here, and deleted automatically once they're used up. You can add/delete the contents of that `external` section using the previously mentioned `cmttools/add-utxo.py` script (see previous section for a brief overview of options).
```

Comment:
`commitments.json` is now in `~/.joinmarket/cmtdata/` by default. Also, similar as above, `cmttools/add-utxo.py` should be changed to `scripts/add-utxo.py`.